### PR TITLE
PHPSecLib code and break out into submodule

### PR DESCRIPTION
**Pull request by mjl3475**
- Still may need to refactor encrypt and decrypt to use better D8 API
- Need to refactor Tests to apply to PHPSecLib

### CODE REVIEW

File: modules/encrypt_seclib/encrypt_seclib.info.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: Encrypt - PHPSecLib Encryption
+description: 'PHPSecLib Encryption for the Encrypt module.'
+core: 8.x
+package: Encryption
+dependencies:
+  - encrypt
```
Comment (nerdstein) [inline modules/encrypt_seclib/encrypt_seclib.info.yml]: Didn't need to use composer manager?

File: modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php [hunk 2]
```diff
@@ -26,8 +27,8 @@ public function getDependencies() {
     if (!\Drupal::moduleHandler()->moduleExists('libraries')) {
       $errors[] = t('You must download and enable the <a href="http://drupal.org/project/libraries">Libraries API</a> module.');
     }
-    if (!file_exists('sites/all/libraries/phpseclib') && !file_exists(\Drupal\Core\DrupalKernel::findSitePath() . '/phpseclib')) {
-      $errors[] = t('You must download the <a href="http://phpseclib.sourceforge.net/">phpseclib</a> library and place it in either sites/all/libraries or sites/yourdomain/libraries.');
+    if (!file_exists('libraries/phpseclib') && !file_exists(\Drupal\Core\DrupalKernel::findSitePath() . '/phpseclib')) {
```
Comment (nerdstein) [inline modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php]: Is this right? does composer automatically load to "libraries"?

File: modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php [hunk 3]
```diff
@@ -42,12 +43,8 @@ public function encrypt($text, $key, $options = array()) {
     $disable_base64 = array_key_exists('base64', $options) && $options['base64'] == FALSE;
 
     if ($path = libraries_get_path('phpseclib')) {
-      // Include the AES file and instantiate.
-      require_once $path . '/Crypt/AES.php';
-      $aes = new Crypt_AES();
-
+      $aes = new AES();
```
Comment (nerdstein) [inline modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php]: Is this the class created by PHPSecLib?

Comment (mjl3475) [inline modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php]: Yes. In the PHP library.

File: modules/encrypt_seclib/src/Tests/EncryptEncryptDecryptTest.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
+<?php
+namespace Drupal\encrypt\Tests;
+
+/**
+ * Test basic encrypting and decripting of a string.
+ * 
+ * @group encrypt
+ */
+class EncryptEncryptDecryptTest extends \Drupal\simpletest\WebTestBase {
```
Comment (nerdstein) [inline modules/encrypt_seclib/src/Tests/EncryptEncryptDecryptTest.php]: Odd name ;) Maybe we can just call it EncryptOperations or something?

File: modules/encrypt_seclib/src/Tests/EncryptPortability.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
+<?php
+namespace Drupal\encrypt\Tests;
+
+/**
+ * Test to make sure an encryption array carries its encryption method and key provider with it to ensure portability.
```
Comment (nerdstein) [inline modules/encrypt_seclib/src/Tests/EncryptPortability.php]: What is this doing? I don't quite get it

File: modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php (hunk 3 above)
Comment (nerdstein) [inline modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php]: Cool ok.

Comment (nerdstein) [inline modules/encrypt_seclib/src/Plugin/EncryptionMethod/PHPSecLibEncryption.php]: I loaded the full file and saw the "use" statement

File: modules/encrypt_seclib/encrypt_seclib.info.yml (hunk 1 above)
Comment (mjl3475) [inline modules/encrypt_seclib/encrypt_seclib.info.yml]: Fixed.

File: modules/encrypt_seclib/src/Tests/EncryptEncryptDecryptTest.php (hunk 4 above)
Comment (mjl3475) [inline modules/encrypt_seclib/src/Tests/EncryptEncryptDecryptTest.php]: Yea, I think we need to revisit testing as a while once we get everything in place.

File: modules/encrypt_seclib/src/Tests/EncryptPortability.php (hunk 5 above)
Comment (mjl3475) [inline modules/encrypt_seclib/src/Tests/EncryptPortability.php]: Not sure, I just copied it over from encrypt until we revisit testing. I think this was pre-existing from the D7 encrypt

**Comment by nerdstein** [comment]
merging

**Pull request merged**